bugbot/rules/file_crash_bug.py

```diff
@@ -43,9 +44,41 @@ def get_bz_params(self, date):
         }
 
     @cached_property
-    def nightly_version(self) -> int:
-        """The version for the nightly channel as defined on Bugzilla."""
-        return utils.get_nightly_version_from_bz()
+    def current_status_flags(self) -> list[str]:
+        """The status flags for Firefox, which includes Nightly, Beta, Release, and ESR."""
+        status = "cf_status_firefox"
+        status_esr = "cf_status_firefox_esr"
+        data: dict[str, list[str]] = {
+            status: [],
+            status_esr: [],
+        }
+
+        def handler(bug):
+            for field in bug.keys():
+                if field.startswith(status_esr):
+                    data[status_esr].append(field)
+                elif field.startswith(status):
+                    data[status].append(field)
+
+        Bugzilla(bugids=1234567, include_fields="_custom", bughandler=handler).wait()
+
+        return (
+            # ESR
+            sorted(
+                data[status_esr],
+                key=lambda flag: int(flag[len(status_esr) :]),
+            )[-1:]
+            # Release, Beta, Nightly
+            + sorted(
+                data[status],
+                key=lambda flag: int(flag[len(status) :]),
+            )[-3:]
+        )
```

Comment:
I wonder if there's a nicer way to do this. CC @calixteman @Archaeopteryx in case you know

Reply:
We do have a different way, which is a bit nicer:

https://github.com/mozilla/bugbot/blob/bf8cad1b18c47d08fa9a00fad1cdb84a3e7f5749/bugbot/bug/analyzer.py#L239-L260



The short comes with that are:
1. It lists two ESR versions, while one of them could be unavailable to set on the bug.
2. It relies on `lmdversions.get()`, which could be inconsistent with Bugzilla around the merge time.

I plan to refactor `analyzer.current_version_flags()` to use the same logic and then call it here. However, I wanted to do it in a separate PR because it is out of the scope and could impact the `regression_set_status_flags` rule.

Reply:
Sounds reasonable!